### PR TITLE
Fix total in `useInfiniteGetList` for empty results

### LINKED ISSUE #9388: useInfiniteGetList returns total as undefined, when real value equal to 0
I have my own dataProvider for getList and I return object like `{ data: [], total: 0 };`

**What you were expecting:**
I am expecting the total value in useListContext is equal to 0.

**What happened instead:**
When total in dataProvider is equal to 0, I receive in useListContext total value as undefined.

There is such code in useInfiniteGetList:
```
return (result.data
	? {
		  ...result,
		  data: result.data,
		  total: result.data?.pages[0]?.total || undefined,
	  }
	: result) as UseInfiniteQueryResult<
	GetInfiniteListResult<RecordType>,
	Error
> & {
	total?: number;
};
```

**Environment**

* React-admin version: 4.15.1
* Last version that did not exhibit the issue (if applicable):
* React version: 18.2.0
* Browser: Chrome
* Stack trace (in case of a JS error):

### PR DESCRIPTION
**Pull request by KislyakovDS**
- total should return 0 if the data is empty;

Closes #9388

### CODE REVIEW

File: packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx [hunk 1]
```diff
@@ -515,4 +515,36 @@ describe('useInfiniteListController', () => {
             });
         });
     });
+
+    describe('checkTotalValue', () => {
```
Comment (djhi) [inline packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx:519]:
```suggestion
```

Not needed

File: packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx [hunk 2]
```diff
@@ -515,4 +515,36 @@ describe('useInfiniteListController', () => {
...
+                const totalDivNode = screen.getByLabelText('Total value');
+                const totalInnerHTML = totalDivNode.innerHTML;
+                const totalValue = Number(totalInnerHTML);
+
+                expect(totalInnerHTML).not.toEqual('undefined');
+                expect(totalValue).not.toBeNaN();
+                expect(totalValue).toEqual(0);
+            });
+        });
+    });
```
Comment (djhi) [inline packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx:549]:
```suggestion
```

**Review by djhi — CHANGES_REQUESTED** [review]
Thanks!

File: packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx (hunk 1 above)
Comment (KislyakovDS) [reply packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx]: done

File: packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx (hunk 2 above)
Comment (KislyakovDS) [reply packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx:547]: done

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**